Sources/CacheAPI-postgres.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines http://www.simplemachines.org
+ * @copyright 2017 Simple Machines and individual contributors
+ * @license http://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 2.1 Beta 3
+ */
+
+if (!defined('SMF'))
+	die('Hacking attempt...');
+
+/**
+ * PostgreSQL Cache API class
+ * @package cacheAPI
+ */
+class postgres_cache extends cache_api
+{
+
+	public function __construct()
+	{
+		parent::__construct();
+
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function connect()
+	{
+		global $db_prefix, $db_connection;
+
+		pg_prepare($db_connection, '', 'SELECT 1 
+			FROM   pg_tables
+			WHERE  schemaname = $1
+			AND    tablename = $2');
+
+		$result = pg_execute($db_connection, '', array('public', $db_prefix . 'cache'));
+
+		if(pg_affected_rows($result) === 0)
+			pg_query($db_connection, 'CREATE TABLE {db_prefix}cache (key text, value text, ttl bigint, PRIMARY KEY (key))');			
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function isSupported($test = false)
+	{
+		global $smcFunc, $db_connection;
+		
+
+		if ($smcFunc['db_title'] !== 'PostgreSQL')
+			return false;
+
+		$result = pg_query($db_connection, 'SHOW server_version_num');
+		$res = pg_fetch_assoc($result);
+		
+		if($res['server_version_num'] < 90500)
+			return false;
+		
+		return true;
+	}
```

Comment:
I think this would work:
```
return $test ? true  parent::isSupported();
```

isSupported handles 2 tests.

First where $test = true, it checks to see if this is a supported API.  This is mainly used for the ACP where we present options for ability to use the API.  We are only seeing if its possible this API is usable, such things as if the function exists.

The second is there $test = false, it does those tests and the actually makes sure its usable.  Such as if cache is enabled, or in the case of memcache it makes sure the settings exist and are valid for it to work.